src/collect-io.cc

```diff
@@ -199,6 +199,7 @@ static gboolean collection_load_private(CollectionData *cd, const gchar *path, C
 	guint flush = !!(flags & COLLECTION_LOAD_FLUSH);
 	guint append = !!(flags & COLLECTION_LOAD_APPEND);
 	guint only_geometry = !!(flags & COLLECTION_LOAD_GEOMETRY);
+	gchar *infotext = nullptr;
```

Comment:
Consider reducing scope and using `g_autofree`.

Reply:
It's in the right scope, it think - because the infotext is needed in the next loop iteration.
g_autofree: +

Reply:
I see, there are a lot of `break` s and `continue`s in loop. Parsing is a bit complex.